src/Chainweb/Mempool/RestAPI/Server.hs

```diff
@@ -28,24 +27,42 @@ import Chainweb.Mempool.Mempool
 import Chainweb.Mempool.RestAPI
 import Chainweb.RestAPI.Orphans ()
 import Chainweb.RestAPI.Utils
+import Chainweb.Time (getCurrentTimeIntegral)
 import Chainweb.Utils
 import Chainweb.Version
 
 ------------------------------------------------------------------------------
-insertHandler :: Show t => MempoolBackend t -> [T.Text] -> Handler NoContent
-insertHandler mempool txsT = handleErrs (NoContent <$ begin)
+insertHandler
+    :: forall t
+    .  ChainwebVersion
+    -> MempoolBackend t
+    -> [T.Text]
+    -> Handler NoContent
+insertHandler v mempool txsT = handleErrs (NoContent <$ begin)
   where
     txcfg = mempoolTxConfig mempool
+
+    decode :: T.Text -> Either String t
     decode = codecDecode (txCodec txcfg) . T.encodeUtf8
 
+    go :: T.Text -> Handler t
     go h = case decode h of
         Left e -> throwM . DecodeException $ T.pack e
         Right t -> return t
 
+    -- | KILLSWITCH: The logic here involving `txActivationDate` can be removed
+    -- once that date itself has passed and Transaction are enabled. Until then,
+    -- this will prevent mempools from inserting data into each other.
+    --
+    begin :: Handler ()
     begin = do
-        txs <- mapM go txsT
-        let txV = V.fromList txs
-        liftIO $ mempoolInsert mempool CheckedInsert txV
+        now <- liftIO getCurrentTimeIntegral
+        case txActivationDate v of
+            Just start | start < now -> pure ()
```

Comment:
@gregorycollins should this throw instead? Do the mempools have certain p2p assumptions (demotion, etc.) when it comes to success/failure when calling this endpoint?